### PR TITLE
Fix subtitle encoder if subrip is requested

### LINKED ISSUE #7958: [Issue]: External Subtitle DeliveryURL Incorrect
### Please describe your bug

Found by investigation in: https://github.com/jellyfin/Swiftfin/issues/417. 

I have `srt` subtitles that are being labeled as `subrip` while using the [DeliveryURL on a MediaStream](https://github.com/jellyfin/jellyfin/blob/0e1a41f7e938e415c32364ff1a4c97b8abab5fae/MediaBrowser.Model/Entities/MediaStream.cs#L446) as the `routeFormat` value will be `subrip` when it expects `srt`. Tested against the API itself in Postman.

Still broken on anticipated `10.8.1` release at time of writing. Logs do not show anything regarding this issue during scanning.

### Jellyfin Version

10.8.0

### if other:

_No response_

### Environment

```markdown
- OS: all
- all others N/A
```

### Jellyfin logs

```shell
[2022-06-17 13:05:30.040 -06:00] [INF] [8] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Movie", Name: "The Big Short", Path: "/Users/epippin/Movies/Jellyfin/Movies/The Big Short/The Big Short.mkv", Id: a55a29fd-4ea5-2d2e-9031-d7472e067363
[2022-06-17 13:05:33.501 -06:00] [INF] [24] Emby.Server.Implementations.Library.Validators.StudiosValidator: Deleting dead "Studio" "7401baee7a70c10835ce0804643a9b44" "Paramount"
[2022-06-17 13:05:33.501 -06:00] [INF] [24] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Studio", Name: "Paramount", Path: "/Users/epippin/.local/share/jellyfin/metadata/Studio/Paramount", Id: 7401baee-7a70-c108-35ce-0804643a9b44
[2022-06-17 13:05:33.502 -06:00] [INF] [24] Emby.Server.Implementations.Library.Validators.StudiosValidator: Deleting dead "Studio" "82547c7f55813279679ba58afac39285" "Plan B Entertainment"
[2022-06-17 13:05:33.502 -06:00] [INF] [24] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Studio", Name: "Plan B Entertainment", Path: "/Users/epippin/.local/share/jellyfin/metadata/Studio/Plan B Entertainment", Id: 82547c7f-5581-3279-679b-a58afac39285
[2022-06-17 13:05:33.503 -06:00] [INF] [24] Emby.Server.Implementations.Library.Validators.StudiosValidator: Deleting dead "Studio" "4ca5f0f35c7228406ca6e16facd71bf6" "Regency Enterprises"
[2022-06-17 13:05:33.503 -06:00] [INF] [24] Emby.Server.Implementations.Library.LibraryManager: Removing item, Type: "Studio", Name: "Regency Enterprises", Path: "/Users/epippin/.local/share/jellyfin/metadata/Studio/Regency Enterprises", Id: 4ca5f0f3-5c72-2840-6ca6-e16facd71bf6
[2022-06-17 13:05:33.505 -06:00] [INF] [24] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Scan Media Library" Completed after 0 minute(s) and 3 seconds
[2022-06-17 13:05:33.509 -06:00] [INF] [24] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks
[2022-06-17 13:05:33.582 -06:00] [INF] [8] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Shows"
[2022-06-17 13:05:33.637 -06:00] [INF] [25] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Music"
[2022-06-17 13:05:33.672 -06:00] [INF] [26] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Movies"
[2022-06-17 13:05:58.006 -06:00] [INF] [23] Emby.Server.Implementations.ScheduledTasks.TaskManager: Executing "Scan Media Library"
[2022-06-17 13:05:58.007 -06:00] [INF] [23] Emby.Server.Implementations.IO.LibraryMonitor: Stopping directory watching for path "/Users/epippin/Movies/Jellyfin/Music"
[2022-06-17 13:05:58.008 -06:00] [INF] [23] Emby.Server.Implementations.IO.LibraryMonitor: Stopping directory watching for path "/Users/epippin/Movies/Jellyfin/Movies"
[2022-06-17 13:05:58.009 -06:00] [INF] [23] Emby.Server.Implementations.IO.LibraryMonitor: Stopping directory watching for path "/Users/epippin/Movies/Jellyfin/Shows"
[2022-06-17 13:05:58.010 -06:00] [INF] [23] Emby.Server.Implementations.Library.LibraryManager: Validating media library
[2022-06-17 13:06:03.373 -06:00] [INF] [108] Emby.Server.Implementations.ScheduledTasks.TaskManager: "Scan Media Library" Completed after 0 minute(s) and 5 seconds
[2022-06-17 13:06:03.400 -06:00] [INF] [108] Emby.Server.Implementations.ScheduledTasks.TaskManager: ExecuteQueuedTasks
[2022-06-17 13:06:03.521 -06:00] [INF] [19] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Music"
[2022-06-17 13:06:03.521 -06:00] [INF] [23] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Shows"
[2022-06-17 13:06:03.521 -06:00] [INF] [15] Emby.Server.Implementations.IO.LibraryMonitor: Watching directory "/Users/epippin/Movies/Jellyfin/Movies"
```

### FFmpeg logs

```
N/A
```

### Please attach any browser or client logs here

N/A

### Please attach any screenshots here

Prior to 10.8.0:
Screen Shot 2022-06-17 at 12 53 49 PM

10.8.0 scan:
Screen Shot 2022-06-17 at 12 45 51 PM
Screen Shot 2022-06-17 at 12 51 03 PM

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shadowghost**
**Changes**
Properly convert subtitles to subrip if requested.

**Issues**
Fixes #7958
Fixes https://github.com/jellyfin/Swiftfin/issues/417

### CODE REVIEW

**Comment by crobibero** [comment]
I would at least add it to the consts that already exist

**Comment by Shadowghost** [comment]
I thought about this too and didn't do it because technically srt and subrio are equivalent and having both of them would imply they aren't...

**Comment by nyanmisaka** [comment]
Can you also take a look at the external ass/ssa subtitle? It doesn’t work for me either.

**Review by crobibero — APPROVED** [review]

**Review by nyanmisaka — APPROVED** [review]

**Comment by Shadowghost** [comment]
> Can you also take a look at the external ass/ssa subtitle? It doesn’t work for me either.

I added barebone writers for SSA and ASS too, so in theory any text sub can be converted to both.

File: MediaBrowser.MediaEncoding/Subtitles/SsaWriter.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                writer.WriteLine();
+                writer.WriteLine("[Events]");
+                writer.WriteLine("Format: Layer, Start, End, Style, Text");
+
+                for (int i = 0; i < trackEvents.Count; i++)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var trackEvent = trackEvents[i];
+                    var timeFormat = @"hh\:mm\:ss\.ff";
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Subtitles/SsaWriter.cs:40]: I would pull the constant strings out of the loop

File: MediaBrowser.MediaEncoding/Subtitles/AssWriter.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                writer.WriteLine();
+                writer.WriteLine("[Events]");
+                writer.WriteLine("Format: Layer, Start, End, Style, Text");
+
+                for (int i = 0; i < trackEvents.Count; i++)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var trackEvent = trackEvents[i];
+                    var timeFormat = @"hh\:mm\:ss\.ff";
```
Comment (crobibero) [inline MediaBrowser.MediaEncoding/Subtitles/AssWriter.cs:40]: same

**Pull request merged**